Submission.java

```diff
@@ -181,23 +258,95 @@ private JSONArray getSprites()
      * Get script count for sprite.
      *
      * @param spriteName 
-     * @return scriptCount 
+     * @return count 
      */
     public int getScriptCountForSprite(String spriteName)
     {
+        return getCountForSprite("scripts", spriteName);
+    }
+    
+    /**
+     * Get variable count for sprite.
+     *
+     * @param spriteName 
+     * @return count 
+     */
+    public int getVariableCountForSprite(String spriteName)
+    {
+        return getCountForSprite("variables", spriteName);
+    }
+    
+    /**
+     * Get list count for sprite.
+     *
+     * @param spriteName 
+     * @return count 
+     */
+    public int getListCountForSprite(String spriteName)
+    {
+        return getCountForSprite("lists", spriteName);
+    }
+    
+    /**
+     * Get script comment count for sprite.
+     *
+     * @param spriteName 
+     * @return count 
+     */
+    public int getScriptCommentCountForSprite(String spriteName)
+    {
+        return getCountForSprite("scriptComments", spriteName);
+    }
+
+    /**
+     * Get sound count for sprite.
+     *
+     * @param spriteName 
+     * @return count
+     */
+    public int getSoundCountForSprite(String spriteName)
+    {
+        return getCountForSprite("sounds", spriteName);
+    }
+
+    /**
+     * Get costume count for sprite.
+     *
+     * @param spriteName 
+     * @return count
+     */
+    public int getCostumeCountForSprite(String spriteName)
+    {
+        return getCountForSprite("costumes", spriteName);
+    }
+    
+    /**
+     * Get count for sprite.
+     *
+     * @param item 
+     * @param spriteName 
+     * @return count 
+     */
+    private int getCountForSprite(String item, String spriteName)
+    {
         JSONArray sprites = getSprites();
-        JSONArray scripts = new JSONArray();
+        JSONArray items = new JSONArray();
         for (int i = 0; i < sprites.size(); i++)
```

Comment:
@KaraBeason Good call. I added the comment to help clarify and changed the 'item' parameter name to 'attribute' so hopefully that makes it a bit more clear. 

I had the same thought about the invalid tests - somewhere along the line I started testing empty attributes, which I think is what I should have been doing for the 'invalid' testing of these functions all along. I went in and removed the useless invalid tests and added tests for empty values where they weren't already implemented. 

Take a look and let me know if you see anything else. Thanks much!